go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -47,6 +51,13 @@ func (bsnc *BaseSecondaryNetworkController) AddSecondaryNetworkResourceCommon(ob
 		}
 		return bsnc.ensurePodForSecondaryNetwork(pod, true)
 
+	case factory.NamespaceType:
+		ns, ok := obj.(*kapi.Namespace)
+		if !ok {
+			return fmt.Errorf("could not cast %T object to *kapi.Namespace", obj)
+		}
+		return bsnc.AddNamespaceForSecondaryNetwork(ns)
+
```

Comment:
didn't think this through yet. But I don't think you need namespace related handling if policy is not based on namespace/pod selector. If so, you don't need to watch for namespace at all. you could simply direct return in WatchNamespaces() without calling WatchResource()

Reply:
@cathy-zhou feel free to keep working exclusively the IPAM case in this PR.

I can tackle the non-IPAM case as a follow-up. 

I think this would help keep this PR "small", otherwise, it'll bloat even more.